libraries/AP_Rally/AP_Rally.h

```diff
@@ -44,8 +42,16 @@ class AP_Rally {
     // data handling
     bool get_rally_point_with_index(uint8_t i, RallyLocation &ret) const;
     bool set_rally_point_with_index(uint8_t i, const RallyLocation &rallyLoc);
-    uint8_t get_rally_total() const { return _rally_point_total_count; }
-    uint8_t get_rally_max(void) const { return _storage.size() / AP_RALLY_WP_SIZE; }
+    uint8_t get_rally_total() const {
+        return uint8_t(_rally_point_total_count);
+    }
+    uint8_t get_rally_max(void) const {
```

Comment:
Isn't this the same thing but more condensed?

`return MIN(_storage.size() / uint16_t(sizeof(RallyLocation)), 255);`